image.go

```diff
@@ -0,0 +1,75 @@
+package canvas
+
+import (
+	"bytes"
+	"image"
+	"image/jpeg"
+	"image/png"
+	"io"
+)
+
+// JPEGImage gives access to the raw bytes.
+// Should be used with PDF, only for baseline JPEGs
+// (progressive might not be displayed properly)
+type JPEGImage interface {
```

Comment:
I don't think we need different structures for JPEG and PNG. We can have one single `Image` struct (like the `bufferedImage` below) with different initializer functions `NewJPEGImage`, `NewPNGImage`, `NewImage`... and add a field specifying the type of image from an enumeration. This will simplify the code.

Also, for what do we need to export the interfaces? If we want to accept images that are already in memory, I suggest we do something as we did for fonts and what the standard library does as well: create the `NewJPEG(io.Reader)` and `NewJPEGBytes([]byte)` functions.